### DISCUSSION TITLE
@click + modifers (e.g. @click.ctrl)

### DISCUSSION THREAD

**Discussion by eckelarsson** [alpinejs/alpine#622] [4. General]
I tried to check the source code to check if this was already enabled but didn't find anything. 
Or does this already work?

Otherwise, I think that would be a good feature to add the ability to listen to ctrl+click, shift+click, option+click, etc etc

(I did clone the source and tried to start on some kind of pull request... But this is to advance for me and I think I'll do more harm than good).

**Comment by SimoTod** [comment]
You are correct. Those modifiers only work with key events at the minute.

**Reply by eckelarsson** [reply]
Thank you for your answer! Do you think it's a good feature to try to implement?

**Reply by SimoTod** [reply]
It could be okay if the implementation doesn't add too much code to the library. I think vue supports them. Probably better to ask Caleb first

**Reply by HugoDF** [reply]
I had a PR that implemented this + a bunch of other modifiers https://github.com/alpinejs/alpine/pull/277 got closed because it added a lot of surface area/bulk to the library

I believe there was an issue (which is now closed since the change got turned down): https://github.com/alpinejs/alpine/issues/273 ~~do mention that you would like to have this in the library~~ do open an issue just for key modifiers without `.exact` maybe, since that linked issue is about `.exact` and not key modifiers

**Reply by eckelarsson** [reply]
Gotcha Hugo, I do think this functionally plays well with the alpinejs. With this implemented, I feel that alpinejs will have a full coverage of click and keyboard events and it still remains small and not overly extended. But I get if the implementation adds to much surface area/bulk.

**Reply by HugoDF** [reply]
Yeah I think so too, totally worth opening an issue, since the other one is about `.exact` 😄

**Reply by ryangjchandler** [reply]
For what's it worth, @HugoDF has an article: https://codewithhugo.com/detect-ctrl-click-js-react-vue/

**Reply by HugoDF** [reply]
Need a new one "Detect ctrl/cmd/meta/shift click in Alpine.js" & a package to go with it 👀
